### DISCUSSION TITLE
channel updates

### DISCUSSION THREAD

**Discussion by bh321** [go-telegram-bot-api/telegram-bot-api#425] [Q&A]
Please show me.
How to remove channel updates or even to automatically remove them from them?
I've seen some Pulls but don't know what to do with them. Just downloading the files didn't solve the problem.
The bot crashes with an error.
Maybe someone can show you a piece of code or more specifically what needs to be done. There is no end to my suffering.

**Comment by tjhorner** [comment]
Could you be a bit more specific? Are you trying to ignore channel updates? What error are you getting when the bot crashes?

**Comment by bh321** [comment]
When a message is sent to the channel where the bot was added, it closes

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x48 pc=0x730bb5]

goroutine 1 [running]:
main.main()
        C:/Users/v108.go:969 +0x285
exit status 2

**Comment by tjhorner** [comment]
You are probably using a field that does not exist in the `Update` or `Message`. You should check that any fields you are using are non-null. A list of optional fields are listed on the [Bot API docs](https://core.telegram.org/bots/api#message). If you still need help, please send a snippet of code surrounding line 969 in your code.

**Comment by bh321** [comment]
the problem occurs on these lines, but only when the message is sent to the channel, if there is no problem in private messages or group chats

} else if update.CallbackQuery.Data != "" {
			if update.CallbackQuery.Data == "123" {

**Reply by Syfaro** [reply]
If `update.CallbackQuery` is nil, trying to access the `Data` field results in a panic. You could change the condition to `update.CallbackQuery != nil && update.CallbackQuery.Data != ""`. I'm not sure if you really need the `Data != ""` part though, unless you're potentially expecting empty data.
